fs/fs.go

```diff
@@ -474,24 +473,22 @@ func (self *RealFsInfo) GetDirInodeUsage(dir string, timeout time.Duration) (uin
 	if dir == "" {
 		return 0, fmt.Errorf("invalid directory")
 	}
-	var counter byteCounter
-	var stderr bytes.Buffer
-	claimToken()
-	defer releaseToken()
-	findCmd := exec.Command("find", dir, "-xdev", "-printf", ".")
-	findCmd.Stdout, findCmd.Stderr = &counter, &stderr
-	if err := findCmd.Start(); err != nil {
-		return 0, fmt.Errorf("failed to exec cmd %v - %v; stderr: %v", findCmd.Args, err, stderr.String())
-	}
-	timer := time.AfterFunc(timeout, func() {
-		glog.Infof("killing cmd %v due to timeout(%s)", findCmd.Args, timeout.String())
-		findCmd.Process.Kill()
+
+	inodes := map[uint64]struct{}{}
+
+	err := filepath.Walk(dir, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return fmt.Errorf("unable to count inodes for part of dir %s: %s", dir, err)
```

Comment:
This matches the previous behavior because `find` would exit non-zero on *most* types of errors you might run into here (notably permission denied), so the err code would be hit before too.

There might be some cases this wouldn't work though.